Makefile

```diff
@@ -0,0 +1,105 @@
+BREW := $(shell which brew)
+BUNDLER := $(shell which bundler)
+DOCKER := $(shell which docker)
+RUBY := $(shell which ruby)
+SWIFT := $(shell which swift)
+SWIFTLINT := $(shell which swiftlint)
+XCODEBUILD := $(shell which xcodebuild)
+
+repo_name := mockingbird
+package_name := Mockingbird
+
+
+# - Build & Set Up
+
+# Builds the project using the Swift package manager
+build:
+	$(SWIFT) build
+.PHONY: build
+
+# Generates an Xcode project for the package
+xcodeproj:
+	$(SWIFT) package generate-xcodeproj --enable-code-coverage
+	$(RUBY) ./Utility/xcodeproj_after_install.rb
+.PHONY: xcodeproj
+
+# Sets up the workspace (typically used after a fresh clone)
+# - Installs dependencies, builds the project, and generates an Xcode project.
+setup: dependencies
+	$(MAKE) build
+	$(MAKE) xcodeproj
+.PHONY: setup
+
+# - Docker
+
+# Builds a docker image with the server
+build-docker:
+	$(DOCKER) build -t $(repo_name) .
+	$(DOCKER) run -it -d -P $(repo_name)
+.PHONY: build-docker
+
+# Starts the server (command should be executed from within the Docker container)
+run-docker:
+	./.build/debug/MockServer
+.PHONY: run-docker
+
+# - Lint
+
+# Lints the source files and generates HTML reports
+lint:
+	mkdir -p Reports
+	$(SWIFTLINT) lint --reporter html > ./Reports/lint_results_swift.html
+	bundle exec rubocop utility/ --format html -o ./Reports/lint_results_ruby.html
```

Comment:
This should be `$(BUNDLE) exec ...` assuming you have a `BUNDLE := $(shell which bundle)`.